### DISCUSSION TITLE
Downgrade Flux version

### DISCUSSION THREAD

**Discussion by ghost** [fluxcd/flux2#2838] [Q&A] [answered]
I installed flux version 0.31.1 and now, i want to downgrade it to version 0.25.2.
How can i do it ?

**Comment by kingdonb** [comment, marked as answer]
The installation process is idempotent and based on the version of the CLI that you used. So simply download the CLI binary for 0.25.2 and use that version to install (bootstrap) again.

However, I will provide the same caution I provided here: https://github.com/fluxcd/flux2/discussions/2840#discussioncomment-2955551 the version you're suggesting to use has CVEs recorded against it.

You may also get some errors if you have been using the latest CRD versions, as I'd expect you must be using if you are on the latest Flux release and the version which you are downgrading to does not have those versions. We recommend using at least 0.31.x now.
